ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/editorhook/js/replacehook.js

```diff
@@ -13,45 +13,74 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  ******************************************************************************/
+'use strict';
 (function (window, author, Coral, channel) {
 
-    const fieldTypes = {BINARY: 'binary', TEXT: 'text', SELECT: 'select', LIST: 'list', DATE: 'date', NON_INPUT: 'nonInputReadOnly'}
+    const fieldTypes = {
+        BINARY: 'binary',
+        TEXT: 'text',
+        SELECT: 'select',
+        LIST: 'list',
+        DATE: 'date',
+        NON_INPUT: 'nonInputReadOnly',
+        CONTAINER: 'container'
+    }
     const typeMap = {
-        'button': fieldTypes.TEXT,
-        'checkboxgroup': fieldTypes.SELECT,
-        'datepicker': fieldTypes.TEXT,
-        'dropdown': fieldTypes.SELECT,
-        'emailinput': fieldTypes.TEXT,
-        'numberinput': fieldTypes.TEXT,
-        'radiobutton': fieldTypes.SELECT,
-        'reset': fieldTypes.TEXT,
-        'submit': fieldTypes.TEXT,
-        'telephoneinput': fieldTypes.TEXT,
-        'text': fieldTypes.NON_INPUT,
-        'textbox': fieldTypes.TEXT,
-        'textinput': fieldTypes.TEXT,
+        'button': fieldTypes.NON_INPUT,
+        'checkbox-group': fieldTypes.SELECT,
+        'date-input': fieldTypes.TEXT,
+        'drop-down': fieldTypes.SELECT,
+        'email': fieldTypes.TEXT,
+        'number-input': fieldTypes.TEXT,
+        'radio-group': fieldTypes.SELECT,
+        'reset': fieldTypes.NON_INPUT,
+        'submit': fieldTypes.NON_INPUT,
+        'text-input': fieldTypes.TEXT,
+        'plain-text': fieldTypes.NON_INPUT,
         'title': fieldTypes.NON_INPUT,
-        'image': fieldTypes.NON_INPUT
+        'image': fieldTypes.NON_INPUT,
+        'panel': fieldTypes.CONTAINER
     }
 
     const preservedProperties = ['id', 'description', 'enabled', 'jcr:created', 'jcr:title', 'name',
         'placeholder', 'readOnly', 'required', 'tooltip', 'visible', 'enum', 'enumNames'];
 
-    const cannotBeReplacedWith = ['fileinput'];
+    const cannotBeReplacedWith = ['file-input'];
 
-    const irreplaceable = ['fileinput'];
+    const irreplaceable = ['file-input'];
 
     const doReplace = window.CQ.FormsCoreComponents.editorhooks.doReplace;
+    const allowedCompFieldTypes = window.CQ.FormsCoreComponents.editorhooks.allowedCompFieldTypes;
+
+
+    function getEditableType(editable) {
+        var result = $.ajax({
+            type: 'GET',
+            async: false,
+            url: Granite.HTTP.externalize(editable.path + ".model.json"),
+            cache: false
+        });
+        return result.responseJSON.fieldType;
+    }
+
+    function getComponentType(compTemplatePath) {
+        var result = $.ajax({
+            type: 'GET',
+            async: false,
+            url: Granite.HTTP.externalize(compTemplatePath + ".json"),
```

Comment:
Why do we have two function to return the same thing ? Can we align this to use the same API